Code/Engine/Core/Messages/EventMessage.h

```diff
@@ -80,17 +77,19 @@ class ezEventMessageSender : public ezMessageSenderBase<EventMessageType>
 
 private:
   EZ_ALWAYS_INLINE void UpdateMessageAndCachedReceiver(ezEventMessage& msg, const ezComponent* pSenderComponent,
-                                                       const ezGameObject* pSearchObject) const
+    const ezGameObject* pSearchObject) const
   {
     msg.m_hSenderObject = pSenderComponent->GetOwner() != nullptr ? pSenderComponent->GetOwner()->GetHandle() : ezGameObjectHandle();
     msg.m_hSenderComponent = pSenderComponent->GetHandle();
 
     if (*reinterpret_cast<ezUInt64*>(&m_hCachedReceiver) == 0xFFFFFFFFFFFFFFFF)
     {
-      m_hCachedReceiver = ezInternal::EventMessageSenderHelper::FindReceiver(msg, pSenderComponent, pSearchObject);
+      if (const ezComponent* pReceiver = pSenderComponent->GetWorld()->FindEventMsgHandler(msg, pSearchObject))
```

Comment:
You need to set the m_hCachedReceiver to an invalid component handle if no component is found, otherwise it will try to find an event handler component again and again.

Reply:
done